src/datasets/table.py

```diff
@@ -980,32 +951,26 @@ def array_cast(array: pa.Array, pa_type: pa.DataType, allow_number_to_str=True):
         if pa.types.is_struct(pa_type) and (
             set(field.name for field in pa_type) == set(field.name for field in array.type)
         ):
-            arrays = [
-                _c(array.field(field.name), field.type, allow_number_to_str=allow_number_to_str) for field in pa_type
-            ]
-            return pa.StructArray.from_arrays(arrays, fields=list(pa_type))
+            arrays = [_c(array.field(field.name), field.type) for field in pa_type]
```

Comment:
Why did you remove `allow_number_to_str` here ?

Reply:
Because we already capture its value here: https://github.com/huggingface/datasets/blob/f013aff823b831fb48b458a7bbb3bb11c2981326/src/datasets/table.py#L974

I would expect an error for passing the same arg twice, but `partial`  supports this due to this line: https://github.com/python/cpython/blob/9d9dc59d07d51d73e5af7dd506d0da63aa336995/Lib/functools.py#L300

Reply:
Ok cool !